src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1432,22 +1438,94 @@ private void LoadAndValidateNetModuleAttributes(ref CustomAttributesBag<CSharpAt
                 }
 
                 diagnostics.Free();
-            }
 
-            Debug.Assert(lazyNetModuleAttributesBag.IsSealed);
+                Debug.Assert(lazyNetModuleAttributesBag.IsSealed);
+            }
         }
 
-        private void EnsureNetModuleAttributesAreBound()
+        private CommonAssemblyWellKnownAttributeData GetLimitedNetModuleDecodedWellKnownAttributeData(QuickAttributes attributeMatches)
         {
-            if (_lazyNetModuleAttributesBag == null)
+            Debug.Assert(attributeMatches is QuickAttributes.AssemblyKeyFile
+                or QuickAttributes.AssemblyKeyName
+                or QuickAttributes.AssemblySignatureKey);
+
+            if (_compilation.Options.OutputKind.IsNetModule())
             {
-                LoadAndValidateNetModuleAttributes(ref _lazyNetModuleAttributesBag);
+                return null;
+            }
+
+            var discardedDiagnostics = BindingDiagnosticBag.Discarded;
```

Comment:
>var discardedDiagnostics = BindingDiagnosticBag.Discarded; [](http://example.com/codeflow?start=12&length=58)

Perhaps it is worth checking if _lazyNetModuleAttributesBag is already not null and return it then.

Reply:
Thanks. Added check to caller